src/bio/bio_buffer.c

```diff
@@ -510,6 +515,69 @@ chunk_get_idle(struct bio_dma_buffer *bdb, struct bio_dma_chunk **chk_ptr)
 	return 0;
 }
 
+static unsigned int
+consumed_pgs(struct bio_dma_buffer *bdb)
+{
+	return (bdb->bdb_tot_cnt * bio_chk_sz) + bdb->bdb_huge_pgs;
+}
+
+static int
+chunk_get_huge(struct bio_dma_buffer *bdb, unsigned int pg_cnt,
+	       struct bio_dma_chunk **chk_ptr)
+{
+	struct bio_dma_chunk	*chk;
+	unsigned int		 tot_pgs = bio_chk_cnt_max * bio_chk_sz;
+	unsigned int		 reclaim_chks, reclaim_pgs;
+	int			 rc;
+
+	D_ASSERT(bio_chk_sz > 0);
+	/* Supported maximum IOV size is 1/2 DMA buffer upper bound size */
+	if (pg_cnt >= (tot_pgs / 2))
+		return -DER_REC2BIG;
```

Comment:
does it mean that it will fail the IO if io_size > 512MB for this VOS tgt? (tot_pgs is "128 * 8MB / PG_SIZE"?)

Reply:
Yes, the minimal total size is 1GB, and it's configured as 1GB by default as well, so >= 512MB value isn't supported.